src/utils.ts

```diff
@@ -17,14 +17,34 @@ export class Utils {
         return collection && typeInstance
             ? (Object.keys(collection)
                   .filter((key: string) => this.hasAllPropertiesOf(collection[key], typeInstance))
-                  .map((key: string) => collection[key]) as T[])
+                  .map((key: string) => this.parseDates(collection[key])) as T[])
             : [];
     }
 
     public static getHrMilliseconds(times: number[]): number {
         return times[0] * 1000 + times[1] / 1e6;
     }
 
+    public static hasOwnProperty<X extends {}, Y extends PropertyKey>(
+        obj: X,
+        prop: Y
+    ): obj is X & Record<Y, unknown> {
+        // informs TS that an object has a property
+        return obj.hasOwnProperty(prop);
+    }
+
+    public static parseDates(obj: unknown): unknown {
+        if (
+            typeof obj === "object" &&
+            obj !== null &&

```

Comment:
Just curious about the ordering of these. What is `typeof obj` return if `obj` is null?

Reply:
Ah yes, this is an interesting one: `typeof null` returns `"object"`
You can try it out here!: 

https://www.typescriptlang.org/play?#code/MYewdgziA2CmB00QHMAUAXAngB1iAZgARgCu00AlANwBQhhQA